lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -658,7 +658,31 @@ sub overview {
         groups  => $groups,
         until   => $until,
     );
-    my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
+
+    my @latest_jobs = [];
+
+    if (@{$search_args->{groupids}} > 1) {
+        my @composed_search_arch;
+
+        foreach my $groupid (@{$search_args->{groupids}}) {
+            my $groupid_search;
+            $groupid_search->{groupid} = $groupid;
+
+            my $local_search_args;
+            $local_search_args->{group_id} = $groupid;
+            $local_search_args->{build} = $self->schema->resultset('Jobs')->latest_build(%$groupid_search);
+            $local_search_args->{distri} = $search_args->{distri} if (defined($search_args->{distri}));
+            $local_search_args->{version} = $search_args->{version} if (defined($search_args->{version}));
+
+            push(@composed_search_arch, $local_search_args);
+        };
+
+        @latest_jobs = $self->schema->resultset('Jobs')->search(\@composed_search_arch)->latest_jobs_multiple_groups($until);
```

Comment:
Not using `compex_query` here means we drop all the ~~complex~~fancy features it provides. Maybe that's good enough but we should check which use cases are broken by this simplification.

Reply:
Remains the investigation of this part

Reply:
maybe https://github.com/os-autoinst/openQA/pull/3918 can help to make complex_query a little bit simpler

Reply:
@ilausuch Is this still open? Or did you address this in your most recent changes?

Reply:
Waiting for #3918 to test if complex_query could help to do it better

Reply:
@ilausuch #3918 is ready to be reviewed